### PR TITLE
Add klicky probe support

### PR DESCRIPTION
**Pull request by Farva42**
Add klicky probe support including stow/deploy menu.

### Description

Add new probe type from Klicky probes like the eluced, quickdraw, klackender, or other probes that require movements to deploy and stow the probe. Allows for probe to be used to home Z.

### Requirements

A probe that is deployed and stowed via movement.

### Benefits

Allows more probes to be used, and allows probes attached with movement to be used to home Z if they are able to.

### Related Issues
#24003
#24370

### CODE REVIEW

File: Marlin/Configuration.h [hunk 1]
```diff
@@ -1243,6 +1282,9 @@
 // For Z_PROBE_ALLEN_KEY see the Delta example configurations.
 //
 
+//Add probe deploy and stow options to the motion menu.
+//#define PROBE_DEPLOY_STOW_MENU
```
Comment (thisiskeithb) [inline Marlin/Configuration.h]: When `BLTOUCH` is enabled, the menu items for deploy & stow are automatically enabled. I'd suggest this menu should also be enabled automatically for these new probes.

**Comment by DerAndere1** [comment]
With a little bit of tweaking this could also make the `SAFE_BED_LEVELING_START_X` ... settings obsolete.

**Comment by Farva42** [comment]
Well that's not going to work.

File: Marlin/Configuration.h (hunk 1 above)
Comment (Farva42) [reply Marlin/Configuration.h]: I updated to make the default behaviour to enable the menu items with the option to disable.

**Comment by thinkyhead** [comment]
Always create a new branch for PRs so you don't pollute your default branch. Also, your bugfix-2.1.x branch is protected so PRs submitted from it cannot be fixed up.

Please merge https://github.com/Farva42/Marlin/pull/3 to update this PR.

**Comment by Farva42** [comment]
My apologies. This is my first time using GitHub and doing a PR definitely learned a few things for next time.

**Comment by nionio6915** [comment]
@thinkyhead what can Euclid team do to assist? 
We've had Euclids running under Marlin for over a year despite the mixed stow/deploy behavior for G29, G35 and G28 gcodes. 
Is this commit going to respond to M401 / M402 calls as well? 
How can we takle the coordination of start/end points of movments next?

**Comment by Farva42** [comment]
This accepts m401 and m402 commands. It also puts a deploy and Stow option in the motion menus(these use m401& m402)

I am going to start looking at how to clean up the behaviour with g28 followed by g29 as it will do a stow and deploy in between. It might take me a while as I'm not the experienced with coding. I've got an idea if what might be causing the inconsistency in positioning after Stow as well.

**Comment by nionio6915** [comment]
Look at issue #22794

We reported it and have tried to trace the source of all the differences.

**Comment by nionio6915** [comment]
How does this play with the G30 bug negating the E parameter?

**Pull request merged**